### PR TITLE
Implement EnqueueExtensions interface in volumerestrictions and volumezone.

### PR DESCRIPTION
**Pull request by whypro**
#### What type of PR is this?

/kind feature
/sig scheduling

#### What this PR does / why we need it:

Implement EnqueueExtensions interface in volumerestrictions and volumezone.

#### Special notes for your reviewer:

Ref #94009

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: pkg/scheduler/framework/plugins/volumerestrictions/volume_restrictions.go [hunk 1]
```diff
@@ -130,6 +131,15 @@ func (pl *VolumeRestrictions) Filter(ctx context.Context, _ *framework.CycleStat
 	return nil
 }
 
+// EventsToRegister returns the possible events that may make a Pod
+// failed by this plugin schedulable.
+func (pl *VolumeRestrictions) EventsToRegister() []framework.ClusterEvent {
+	return []framework.ClusterEvent{
+		{Resource: framework.Pod, ActionType: framework.Delete | framework.Update},
```
Comment (ahg-g) [inline pkg/scheduler/framework/plugins/volumerestrictions/volume_restrictions.go:138]: lets add a comment on each explaining how they impact schedulability of pods.

Comment (ahg-g) [reply pkg/scheduler/framework/plugins/volumerestrictions/volume_restrictions.go:138]: @cofyc can you help advise?

File: pkg/scheduler/framework/plugins/volumezone/volume_zone.go [hunk 2]
```diff
@@ -171,6 +172,17 @@ func (pl *VolumeZone) Filter(ctx context.Context, _ *framework.CycleState, pod *
 	return nil
 }
 
+// EventsToRegister returns the possible events that may make a Pod
+// failed by this plugin schedulable.
+func (pl *VolumeZone) EventsToRegister() []framework.ClusterEvent {
```
Comment (ahg-g) [inline pkg/scheduler/framework/plugins/volumezone/volume_zone.go:177]: ditto, lets add a comment on each explaining how they impact schedulability of pods.

cc @cofyc

File: pkg/scheduler/framework/plugins/volumerestrictions/volume_restrictions.go (hunk 1 above)
Comment (ahg-g) [reply pkg/scheduler/framework/plugins/volumerestrictions/volume_restrictions.go:138]: see example: https://github.com/kubernetes/kubernetes/pull/100003

Comment (whypro) [reply pkg/scheduler/framework/plugins/volumerestrictions/volume_restrictions.go:138]: ok

Comment (whypro) [reply pkg/scheduler/framework/plugins/volumerestrictions/volume_restrictions.go]: Thanks for your suggestion. I've added inline comments with the latest push.

File: pkg/scheduler/framework/plugins/volumezone/volume_zone.go [hunk 3]
```diff
@@ -171,6 +172,23 @@ func (pl *VolumeZone) Filter(ctx context.Context, _ *framework.CycleState, pod *
...
+	return []framework.ClusterEvent{
+		// New storageClass add with bind mode `VolumeBindingWaitForFirstConsumer` will make a pod schedulable.
+		// Due to immutable fields `storageClass.volumeBindingMode`, storageClass update events are ignored.
```
Comment (cofyc) [inline pkg/scheduler/framework/plugins/volumezone/volume_zone.go:180]:
```suggestion
		// Due to immutable field `storageClass.volumeBindingMode`, storageClass update events are ignored.
```

File: pkg/scheduler/framework/plugins/volumezone/volume_zone.go [hunk 4]
```diff
@@ -171,6 +172,23 @@ func (pl *VolumeZone) Filter(ctx context.Context, _ *framework.CycleState, pod *
 	return nil
 }
 
+// EventsToRegister returns the possible events that may make a Pod
+// failed by this plugin schedulable.
+func (pl *VolumeZone) EventsToRegister() []framework.ClusterEvent {
+	return []framework.ClusterEvent{
+		// New storageClass add with bind mode `VolumeBindingWaitForFirstConsumer` will make a pod schedulable.
```
Comment (cofyc) [inline pkg/scheduler/framework/plugins/volumezone/volume_zone.go:179]:
```suggestion
		// New storageClass with bind mode `VolumeBindingWaitForFirstConsumer` will make a pod schedulable.
```

**Comment by cofyc** [comment]
registered events LGTM

**Comment by whypro** [comment]
/test pull-kubernetes-e2e-gce-ubuntu-containerd
--

**Pull request merged**